Fixes scrambled error output when call to retrieve script is nested

diverdane opened the pull request:
### What does this PR do?

This change fixes an issue whereby:
- If the `0001_retrieve-secrets.sh` script is being sourced/called from another script (as is done in production)
- And an error occurs in parsing the variable settings in the `conjur-env` binary output (e.g. for an invalid variable name)
- Then the fields of error messages being generated by the `0001_retrieve-secrets.sh` script were getting scrambled and the error message was getting concatenated multiple times.

Here is an example of an error that was getting generated:

```
./tmp/0001_retrieve-secrets.sh: Error on line ./tmp/0001_retrieve-secrets.sh: Unable to export `/tmp/bats-run-1/bats.111.src`; value may not be a valid identifier/opt/bats/lib/bats-core/test_functions.bash: Error on line /tmp/bats-run-1/bats.111.src: Unable to export `/opt/bats/libexec/bats-core/bats-exec-test`; value may not be a valid identifier/opt/bats/libexec/bats-core/bats-exec-test: Error on line 74: Unable to export `ANOTHER_BAD.VAR`; value may not be a valid identifier
```

The source of the problem is that the error message printf format was using
an expanded Bash array (`BASH_SOURCE[@]`) (rather than the first
element) to display the script name, and the elements of the array were
then being used for subsequent printf fields.

### What ticket does this PR close?
Resolves #120

### Checklists

#### Change log
- [x] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [x] This PR includes new unit and integration tests to go with the code changes, or
- [ ] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [x] This PR does not require updating any documentation

izgeri commented:
can you please add a changelog entry? it's ok to have it link to the PR, since there's no issue right now - but it might be best to file a quick issue. thanks for digging into this!

diverdane commented:
@izgeri I added a CHANGELOG entry. There was an issue (#120) to use for this.
I also simplified the UT for this, since only one level of nesting is required for repro, and that's being done already in every test case.
